sources/csv/README.md

```diff
@@ -0,0 +1,122 @@
+# HarbourBridge: CSV-to-Spanner Migration
+
+HarbourBridge is a stand-alone open source tool for Cloud Spanner evaluation 
+and migration. We now support loading data from CSVs. This assumes a Spanner
+database with schema already exists and HarbourBridge loads the data for you.
+It first reads the schema in the database specified by your target profile
+to understand how to convert the data to relevant types. If using PG Spanner,
+you should specify the dialect in the target-profile explicitly.
+**For CSVs, you need to run harbourbridge in data mode only**
+
+## Example CSV Usage
+
+You can load CSVs to Spanner in 2 ways.
+
+- **Without manifest input:**
+ Before running, make sure each table's csv is present in the current working 
+directory named `[table_name].csv` where table_name is the same as the Spanner
+table name. 
+Harbourbridge will migrate `table_name.csv` to a Spanner table named
+`table_name` in the database specified via the target profile.
+
+```sh
+harbourbridge data -source=csv -target-profile="instance=my-instance,dbname=my-db,dialect=postgresql" 
+```
+
+- **Providing a manifest input:**
+ You can optionally provide a manifest file if you have csv files in different
+locations (local system as well as Google Cloud Storage). You can also provide 
+multiple csv file paths for a single table using the manifest.
+
+```sh
+harbourbridge data -source=csv -source-profile="manifest=path/to/manifest/file" -target-profile="instance=my-instance,dbname=my-db" 
+```
+
+### Manifest File
+The manifest file should be a list of JSONs containing each table's
+csv file locations. Each item should contain the fields:
+- `"table_name"`: The name of the table, identical to the corresponding table 
+name in your Spanner schema.
+- `"file_patterns"`: A list of the CSV file paths (local and GCS) that contain 
+data for that table. Note that it is a list, so multiple CSV files are accepted 
+per table.
+
+**Sample manifest:**
+```
+[
+    {
+      "table_name": "Albums",
+      "file_patterns": [
+        "/Users/username/Desktop/Albums.csv"
+      ]
+    },
+    {
+      "table_name": "Singers",
+      "file_patterns": [
+        "/Users/username/Desktop/tmp/Singers_1.csv",
+        "gs://bucket-name/Singers_2.csv",
+        "/Users/username/Downloads/other/Singers_3.csv"
+      ]
+    }
+]
+```
+**CAVEATS:**
+- File patterns do not accept regular expressions. Provide the path inside 
+double quotes.
+
+### CSV File Format
+- Harbourbridge checks the first row and matches it with the spanner columns
+to check if the first row is a permutation of the Spanner table's column names.
+If a match is not found, then Harbourbridge assumes the order of data in csv
```

Comment:
do we have sanity checks to see the number of columns match ?

Reply:
Good catch, I missed this check. Added